etc/eslint/parser-options/typescript.js

```diff
@@ -0,0 +1,57 @@
+/**
+* @license Apache-2.0
+*
+* Copyright (c) 2023 The Stdlib Authors.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+'use strict';
+
+// MODULES //
+
+var path = require( 'path' );
+var rootDir = require( './../../../lib/node_modules/@stdlib/_tools/utils/root-dir' );
+
+
+// MAIN //
+
+/**
+* ESLint parser options.
+*
+* @namespace options
+*/
+var options = {};
+
+/**
+* The project's TypeScript configuration file.
+*
+* @name project
+* @memberof options
+* @type {string}
+*/
+options.project = path.join( rootDir(), 'tsconfig.json' );
```

Comment:
@Planeshifter Instead of having the `tsconfig.json` in the root directory, can we move to `./etc` now? We previously had in the root directory due to tooling limitations, but, if we can have it in any arbitrary location, it would be nicer if we can move to `./etc`. And if in `etc`, I think we can avoid needing to resolve the root directory.

As it is, it seems a little silly to walk the directory tree to grab a package to resolve the root dir, when doing so basically assumes the root directory (!). In either case, this is brittle and could break if things are moved around. So might as well make it explicit when resolving the path of the actual file we are interested in.

Reply:
@Planeshifter This still needs to be resolved. Namely, we don't need to use `rootDir()` given that we are already walking the directory tree.

Reply:
@Planeshifter This still needs to be resolved.

Reply:
@kgryte As mentioned in the discussion of whether we can move `tsconfig.json` to `etc/`, this breaks VSCode linting of TS files for me as it necessitates updating the `include` glob patterns of the `tsconfig.json` file. But using e.g.

```
    "**/docs/types/index.d.ts",
    "**/docs/types/test.ts",
    "**/lib/node_modules/**/docs/types/index.d.ts",
    "**/lib/node_modules/**/docs/types/test.ts",
    "**/lib/node_modules/**/@stdlib/types/index.d.ts",
    "**/lib/node_modules/**/@stdlib/types/test.ts"
```

does not work, likely because by default everything in `node_modules` folders is ignored automatically and we need to have a more specific include to overrule it and the above does not count.

 Similarly, I had to use `rootDir()` for the `tsconfig.json` file to get resolved correctly. Let me know what else you would put here and I can confirm whether it works or breaks VSCode linting for me but this was chosen deliberatively to make it work.

Reply:
@Planeshifter The point is that you don't need to use `rootDir`. The path of the require statement to resolve `root-dir` already makes an assumption about where the root project directory is. You have to walk up to the root dir and then down to get the desired package. And then the package is used to resolve the root directory. That is resolution is redundant, as you are already manually resolving the root directory.

Reply:
@Planeshifter Since this doesn't seem to be clear, examine a bit what this line is doing:

```javascript
var rootDir = require( './../../../lib/node_modules/@stdlib/_tools/utils/root-dir' );
```

Reply:
Also, if what you are needing is the absolute file path, that is not clear from the implementation. However, that can be achieved, IIRC, by doing something along the lines of

```javascript
path.resolve( __dirname, '..', '..', '..', 'tsconfig.json' )
```

But then, I suppose this may be problematic during bundling.

In short, I'm asking why `@stdlib/_tools/utils/root-dir` is necessary here. Currently, the code includes no explanation or rationale, nor has one been provided in this PR, apart from the handwavy "to make it work".

Currently, it looks to me as though the usage of this package is superfluous.

Reply:
@kgryte We need an absolute path as for some reason relative path resolution doesn't seem to work properly. So yes, I could have used `path.resolve( __dirname, '..', '..', '..', 'tsconfig.json' )` as well.

Reply:
@Planeshifter Would you mind adding a comment to the code? Otherwise, future me is likely to forget why using `root-dir` is necessary and not understand the basis for the current approach.